Add a total summary and always show all tests.

czunker opened the pull request:
Whether the total summary is shown can be specified with an option.

Fixes #528

Signed-off-by: Christian Zunker <christian.zunker@codecentric.cloud>

lizrice commented on cmd/common.go:
```diff
@@ -159,7 +162,7 @@ func prettyPrint(r *check.Controls, summary check.Summary) {
 	if !noRemediations {
 		if summary.Fail > 0 || summary.Warn > 0 {
-			colors[check.WARN].Printf("== Remediations ==\n")
+			colors[check.WARN].Printf("== Remediations %s==\n", r.Type)
```
Nice addition

lizrice commented on cmd/master.go:
```diff
@@ -34,7 +34,7 @@ var masterCmd = &cobra.Command{
 
 		filename := loadConfig(check.MASTER, bv)
-		runChecks(check.MASTER, filename)
+		runChecks(check.MASTER, filename, !isMaster())
```
This should always attempt to run regardless of node type, because the user has explicitly asked for the master tests

lizrice commented on cmd/node.go:
```diff
@@ -34,7 +34,7 @@ var nodeCmd = &cobra.Command{
 
 		filename := loadConfig(check.NODE, bv)
-		runChecks(check.NODE, filename)
+		runChecks(check.NODE, filename, false)
```
Same comment as for master, if the user expicitly tries to run the node test cases we should make an attempt

lizrice commented on cmd/root.go:
```diff
@@ -71,35 +72,35 @@ var RootCmd = &cobra.Command{
-				glog.V(1).Info("== Running control plane checks ==")
-				runChecks(check.CONTROLPLANE, loadConfig(check.CONTROLPLANE, bv))
-			}
+		isMasterNode := isMaster()
```
Can you explain your thinking on changing this logic here? I think I preferred all the decisions about whether to skip a section here in one place, but perhaps I am missing something?

lizrice reviewed (commented):
Thanks for your work on this so far! I also commented on #528 with a suggestion about how to present the sections that are skipped.

lizrice commented on cmd/root.go:
```diff
@@ -86,6 +87,8 @@ var RootCmd = &cobra.Command{
 				glog.V(1).Info("== Running control plane checks ==")
 				runChecks(check.CONTROLPLANE, loadConfig(check.CONTROLPLANE, bv))
 			}
+		} else {
+			fmt.Println("== Skipping master checks ==")
```
Let's keep these as log output

lizrice reviewed (changes requested):
I'd just like to change some of the stdout output into logs (see the suggestions) but otherwise this looks good, thank you!

lizrice commented:
@yoavrotems @danielpacak note that when we merge this, it will change the default JSON output to include overall totals

czunker commented:
Hi @lizrice ,
changed the output according to your suggestions.

yoavrotems commented:
@czunker Hey I fixed some conflicts and now the PR seems to fail the tests

czunker commented:
Hi @yoavrotems ,

rebased to master and fixed the first issue.
It seems, there was a closing bracket missing:
https://github.com/aquasecurity/kube-bench/pull/759/commits/75e5aa9c0a4b5f260b7a1038f5775d31e703e78d#diff-f3346ecefc99144b92b77b4e6ae336aafbd52a75b9587e026ce0325162fc7afdR709

No some results fail in the integration tests. Correcting the numbers. That just than pass the integration test.

lizrice reviewed (approved):
lgtm
